<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd"><html xmlns="http://www.w3.org/1999/xhtml"><head>
<title>src/javax/naming/NamingEnumeration.java</title>
<meta name="robots" content="noindex,nofollow">
<meta name="generator" content="GLOBAL-6.6.3">
<meta http-equiv="Content-Style-Type" content="text/css">
<link rel="stylesheet" type="text/css" href="../style.css">
</head>
<body>
<a id="TOP" name="TOP"></a><div class="comment">
/* [&lt;][&gt;]<a href="#L105">[^]</a><a href="#L146">[v]</a>[top]<a href="#BOTTOM">[bottom]</a><a href="../mains.html">[index]</a><a href="../help.html">[help]</a> */</div>
<hr>
<h2 class="header">DEFINITIONS</h2>
This source file includes following definitions.
<ol>
<li><a href="#L105" title="Defined at 105.">next</a></li>
<li><a href="#L121" title="Defined at 121.">hasMore</a></li>
<li><a href="#L146" title="Defined at 146.">close</a></li>
</ol>
<hr>
<pre>
<div class="comment">
  Copyright (c) 1999, 2004, Oracle and/or its affiliates. All rights reserved.
  ORACLE PROPRIETARY/CONFIDENTIAL. Use is subject to license terms.</div>
<a id="L25" name="L25"></a>  25 
<a id="L26" name="L26"></a>  26 <strong class="reserved">package</strong> <a href="../D/26637.html" title="Multiple defined in 6 places.">javax</a>.naming;
<a id="L27" name="L27"></a>  27 
<a id="L28" name="L28"></a>  28 <strong class="reserved">import</strong> java.util.Enumeration;
<a id="L29" name="L29"></a>  29 
<div class="comment">
   This interface is for enumerating lists returned by
   methods in the javax.naming and javax.naming.directory packages.
   It extends Enumeration to allow as exceptions to be thrown during
   the enumeration.
  <p>
   When a method such as list(), listBindings(), or search() returns
   a NamingEnumeration, any exceptions encountered are reserved until
   all results have been returned. At the end of the enumeration, the
   exception is thrown (by hasMore());
   </p><p>
   For example, if the list() is
   returning only a partial answer, the corresponding exception would
   be PartialResultException. list() would first return a NamingEnumeration.
   When the last of the results has been returned by the NamingEnumeration's
   next(), invoking hasMore() would result in PartialResultException being thrown.
  </p><p>
   In another example, if a search() method was invoked with a specified
   size limit of 'n'. If the answer consists of more than 'n' results,
   search() would first return a NamingEnumeration.
   When the n'th result has been returned by invoking next() on the
   NamingEnumeration, a SizeLimitExceedException would then thrown when
   hasMore() is invoked.
  </p><p>
   Note that if the program uses hasMoreElements() and nextElement() instead
   to iterate through the NamingEnumeration, because these methods
   cannot throw exceptions, no exception will be thrown. Instead,
   in the previous example, after the n'th result has been returned by
   nextElement(), invoking hasMoreElements() would return false.
  </p><p>
   Note also that NoSuchElementException is thrown if the program invokes
   next() or nextElement() when there are no elements left in the enumeration.
   The program can always avoid this exception by using hasMore() and
   hasMoreElements() to check whether the end of the enumeration has been reached.
  </p><p>
   If an exception is thrown during an enumeration,
   the enumeration becomes invalid.
   Subsequent invocation of any method on that enumeration
   will yield undefined results.
   @author Rosanna Lee
   @author Scott Seligman
   @see Context#list
   @see Context#listBindings
   @see javax.naming.directory.DirContext#search
   @see javax.naming.directory.Attributes#getAll
   @see javax.naming.directory.Attributes#getIDs
   @see javax.naming.directory.Attribute#getAll
   @since 1.3</p></div>
<a id="L81" name="L81"></a>  81 <strong class="reserved">public</strong> <strong class="reserved">interface</strong> <a href="../R/5470.html" title="Multiple referred from 63 places.">NamingEnumeration</a>&lt;<a href="../D/8701.html" title="Multiple defined in 3 places.">T</a>&gt; <strong class="reserved">extends</strong> <a href="../S/1134.html#L61" title="Defined at 61 in src/java/util/Enumeration.java.">Enumeration</a>&lt;<a href="../D/8701.html" title="Multiple defined in 3 places.">T</a>&gt; <em class="brace">{</em>
<div class="comment">
       Retrieves the next element in the enumeration.
       This method allows naming exceptions encountered while
       retrieving the next element to be caught and handled
       by the application.
       <p>
       Note that <tt>next()</tt> can also throw the runtime exception
       NoSuchElementException to indicate that the caller is
       attempting to enumerate beyond the end of the enumeration.
       This is different from a NamingException, which indicates
       that there was a problem in obtaining the next element,
       for example, due to a referral or server unavailability, etc.
       @return         The possibly null element in the enumeration.
           null is only valid for enumerations that can return
           null (e.g. Attribute.getAll() returns an enumeration of
           attribute values, and an attribute value can be null).
       @exception NamingException If a naming exception is encountered while attempting
                       to retrieve the next element. See NamingException
                       and its subclasses for the possible naming exceptions.
       @exception java.util.NoSuchElementException If attempting to get the next element when none is available.
       @see java.util.Enumeration#nextElement</p></div>
<a id="L105" name="L105"></a> 105     <strong class="reserved">public</strong> <a href="../D/8701.html" title="Multiple defined in 3 places.">T</a> <a href="../R/23650.html" title="Multiple referred from 3301 places.">next</a>() <strong class="reserved">throws</strong> <a href="../S/4338.html#L60" title="Defined at 60 in src/javax/naming/NamingException.java.">NamingException</a>;
<a id="L106" name="L106"></a> 106 
<div class="comment">
       Determines whether there are any more elements in the enumeration.
       This method allows naming exceptions encountered while
       determining whether there are more elements to be caught and handled
       by the application.
       @return         true if there is more in the enumeration ; false otherwise.
       @exception NamingException
                       If a naming exception is encountered while attempting
                       to determine whether there is another element
                       in the enumeration. See NamingException
                       and its subclasses for the possible naming exceptions.
       @see java.util.Enumeration#hasMoreElements</div>
<a id="L121" name="L121"></a> 121     <strong class="reserved">public</strong> <strong class="reserved">boolean</strong> <a href="../R/20226.html" title="Multiple referred from 35 places.">hasMore</a>() <strong class="reserved">throws</strong> <a href="../S/4338.html#L60" title="Defined at 60 in src/javax/naming/NamingException.java.">NamingException</a>;
<a id="L122" name="L122"></a> 122 
<div class="comment">
      Closes this enumeration.
      After this method has been invoked on this enumeration, the
      enumeration becomes invalid and subsequent invocation of any of
      its methods will yield undefined results.
      This method is intended for aborting an enumeration to free up resources.
      If an enumeration proceeds to the end--that is, until
      <tt>hasMoreElements()</tt> or <tt>hasMore()</tt> returns <tt>false</tt>--
      resources will be freed up automatically and there is no need to
      explicitly call <tt>close()</tt>.
     <p>
      This method indicates to the service provider that it is free
      to release resources associated with the enumeration, and can
      notify servers to cancel any outstanding requests. The <tt>close()</tt>
      method is a hint to implementations for managing their resources.
      Implementations are encouraged to use appropriate algorithms to
      manage their resources when client omits the <tt>close()</tt> calls.
      @exception NamingException If a naming exception is encountered
      while closing the enumeration.
      @since 1.3</p></div>
<a id="L146" name="L146"></a> 146     <strong class="reserved">public</strong> <strong class="reserved">void</strong> <a href="../R/11587.html" title="Multiple referred from 410 places.">close</a>() <strong class="reserved">throws</strong> <a href="../S/4338.html#L60" title="Defined at 60 in src/javax/naming/NamingException.java.">NamingException</a>;
<a id="L147" name="L147"></a> 147 <em class="brace">}</em>
</pre>
<hr>
<div class="comment">
/* [&lt;][&gt;]<a href="#L105">[^]</a><a href="#L146">[v]</a><a href="#TOP">[top]</a>[bottom]<a href="../mains.html">[index]</a><a href="../help.html">[help]</a> */</div>
</body></html>